Fix issue #135

ezbr opened the pull request:
Fixed https://github.com/google/personfinder/issues/135 by counting series of CJK characters as one word.

gimite commented on app/results.py:
```diff
@@ -51,6 +51,19 @@ def is_possible_phone_number(query_str):
 
+def max_word_length(query_words):
+    max_length = max(map(len, query_words))
+    # Count series of CJK characters as one word.
```
Are you doing this because each CJK character is treated as one word in query_words even if it originally doesn't have whitespace e.g., "山田" -> ["山", "田"]? Have you verified that behavior? If so, can you explain it in the comment here?

ezbr replied:
Yes, I verified that behavior. Comment added.

gimite commented on app/results.py:
```diff
@@ -51,6 +51,19 @@ def is_possible_phone_number(query_str):
+    max_length = max(map(len, query_words))
+    # Count series of CJK characters as one word.
+    cjk_length = 0
+    for i in range(len(query_words)):
```
You can simplify this to "for word in query_words".

ezbr replied:
Done.

gimite commented on app/results.py:
```diff
@@ -51,6 +51,19 @@ def is_possible_phone_number(query_str):
+    cjk_length = 0
+    for i in range(len(query_words)):
+        word = query_words[i]
+        if u'\u3400' <= word <= u'\u9fff':
```
Add a comment that this range is copied from the logic in text_query.py. Otherwise it is a bit confusing because this range actually doesn't represent CJK well :)

ezbr replied:
Comment added.

gimite commented:
Somehow your pull request started to show so many diffs:
https://github.com/google/personfinder/pull/156/files
including recent change to add Babel. Can you try:

    $ git pull origin master
    $ git merge master

on your branch, maybe?

ezbr commented:
Yeah, I don't know what caused that. When I tried those two commands, it
just says "Already up-to-date." If you look at this page, you can see the
real changes:
https://github.com/google/personfinder/compare/ezbr-fix-issue_135. I think
all the other changes are the changes that have been merged to master since
I created my branch maybe, but I don't know.

On Thu, Sep 10, 2015 at 3:10 PM, Hiroshi Ichikawa <notifications@github.com>
wrote:

> Somehow your pull request started to show so many diffs:
> https://github.com/google/personfinder/pull/156/files
> including recent change to add Babel. Can you try:
>
> $ git pull origin master
> $ git merge master
>
> on your branch, maybe?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/personfinder/pull/156#issuecomment-139126947>.
>

gimite commented:
Can you try "git push origin ezbr-fix-issue_135" after running the commands?

ezbr commented:
I tried, it just says "Everything up-to-date". I think it is showing all
those files because I merged the changes I had on that branch into a git
directory on another computer that had all of these other changes made to
master that didn't exist in the ezbr-fix-issue_135. I'm pretty sure that
the only difference between master and ezbr-fix-issue_135 now are the
changes shown here:
https://github.com/google/personfinder/compare/ezbr-fix-issue_135.

On Thu, Sep 10, 2015 at 4:00 PM, Hiroshi Ichikawa <notifications@github.com>
wrote:

> Can you try "git push origin ezbr-fix-issue_135" after running the
> commands?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/personfinder/pull/156#issuecomment-139133782>.
>

gimite commented:
Hmm I believe it should automatically compare against the master branch you merged last time, but something may be broken.

The change LGTM. Can you pull and merge latest master again and try:

    tools/all_tests

to see if the tests are not broken?

You need to pull again because I have just fixed a few tests. You can ignore PhotoUploadTest failure because it depends on some image libraries.

ezbr commented:
Okay, I merged master and now it correctly shows only the file that I
changed. I also ran tools/all_tests and all unit and server tests passed.

On Mon, Sep 14, 2015 at 4:44 PM, Hiroshi Ichikawa <notifications@github.com>
wrote:

> Hmm I believe it should automatically compare against the master branch
> you merged last time, but something may be broken.
>
> The change LGTM. Can you pull and merge latest master again and try:
>
> tools/all_tests
>
> to see if the tests are not broken?
>
> You need to pull again because I have just fixed a few tests. You can
> ignore PhotoUploadTest failure because it depends on some image libraries.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/personfinder/pull/156#issuecomment-139986364>.
>

gimite commented:
Merged. Thanks!
